packages/network/src/createSyncWorker.ts

```diff
@@ -14,22 +14,39 @@ import { Input, Ack, ack } from "./workers/SyncWorker";
  * dispose: function to dispose of the sync worker
  * }
  */
-export function createSyncWorker<C extends Components = Components>(ack$?: Observable<Ack>) {
+export function createSyncWorker<C extends Components = Components>(
+  ack$?: Observable<Ack>,
+  options?: { thread?: "main" | "worker" }
+) {
+  const thread = options?.thread || "worker";
   const input$ = new Subject<Input>();
-  const worker = new Worker(new URL("./workers/Sync.worker.ts", import.meta.url), { type: "module" });
   const ecsEvents$ = new Subject<NetworkEvent<C>[]>();
+  let dispose: () => void;
 
   // Send ack every 16ms if no external ack$ is provided
   ack$ = ack$ || timer(0, 16).pipe(map(() => ack));
   const ackSub = ack$.subscribe(input$);
 
-  // Pass in a "config stream", receive a stream of ECS events
-  const subscription = fromWorker<Input, NetworkEvent<C>[]>(worker, input$).subscribe(ecsEvents$);
-  const dispose = () => {
-    worker.terminate();
-    subscription?.unsubscribe();
-    ackSub?.unsubscribe();
-  };
+  // If thread option is "worker", create a new web worker to sync the state
+  if (thread === "worker") {
+    const worker = new Worker(new URL("./workers/Sync.worker.ts", import.meta.url), { type: "module" });
+
+    // Pass in a "config stream", receive a stream of ECS events
+    const subscription = fromWorker<Input, NetworkEvent<C>[]>(worker, input$).subscribe(ecsEvents$);
+    dispose = () => {
+      worker.terminate();
+      subscription?.unsubscribe();
+      ackSub?.unsubscribe();
+    };
+  } else {
+    // Otherwise sync the state in the main thread
+    // Pass in a "config stream", receive a stream of ECS events
+    const subscription = new SyncWorker<C>().work(input$).subscribe(ecsEvents$);
+    dispose = () => {
+      subscription?.unsubscribe();
+      ackSub?.unsubscribe();
+    };
```

Comment:
Not blocking here but a potential solution if we find we need to lean on `requestIdleCallback` to avoid long blocks on the main thread: https://ncjamieson.com/how-to-use-requestidlecallback/